broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionTransitionImpl.java

```diff
@@ -109,6 +109,9 @@ private void enqueueNextTransition(
       ongoingTransitionFuture = currentTransitionFuture;
 
       final var ongoingTransition = currentTransition;
+
+      LOG.info(
```

Comment:
:x: This will be logged everytime there is transition. Previous transition needs to be cancelled only if it was not completed. So you should log only if `!ongoingTransition.isCompleted()`

Reply:
Oh nice catch! I didn't realize that `currentTransition` is not cleared when it is completed :+1:

Reply:
May be it is better to clear `currentTransition` after it is completed. Not sure if it causes other issues.

Reply:
It's tricky to do this correctly. By the time a transition finishes, the `currentTransition` might be a newer transition so we shouldn't just set it to null.

Reply:
Yeah, I'd rather not touch that right now so I only added a check so that we only log if the "current" transition is not completed, like you suggested.